cmake/AutodetectLibrary.cmake

```diff
@@ -0,0 +1,164 @@
+#
+# Try to autodetect a library via various of means.
+#
+# We first assume that find_package() can find it.
+# Next we try it via pkg-config.
+# If all that fails, do a find_path()/find_library(), as last effort.
+#
+# This is done as we support a wide variaty of systems, and not all have the
+# CMake files we need. For example, lzo doesn't even have pkg-config files
+# in Debian Stretch (but does in Debian Sid).
+#
+
+macro(_autodetect_library_via_pkgconfig NAME PKGCONFIG)
+    include(FindPkgConfig QUIET)
+
+    if (PKG_CONFIG_FOUND)
+        pkg_search_module("${NAME}" "${PKGCONFIG}" QUIET)
+
+        # pkg-config returns the libraries without really resolving them.
+        # This has as huge drawback that sometimes you only notice on linking
```

Comment:
Maybe you should require a more recent CMake version: newer versions of FindPkgConfig do exactly that. And they return all things like library, compile flags, and include directories in an imported target.

Reply:
I wish we could. I looked for the highest CMake version we needed to support (for now), which is 3.5. But over time we can bump it, and remove these quirks for sure :) Happy to see it is fixed in more recent versions, as it was really weird :P

Reply:
What's the problem? Kitware provides statically linked binaries for at least Windows and Linux and an Ubuntu repo. Someone wants to build your thing with a ton of external dependencies, getting a recent CMake version shoudn't be a big deal.